pkg/apis/compliance/v1alpha1/customrule_types.go

```diff
@@ -0,0 +1,132 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ScannerType string
+
+const (
+	ScannerTypeCEL      ScannerType = "CEL"
+	ScannerTypeOpenSCAP ScannerType = "OpenSCAP"
+	ScannerTypeUnknown  ScannerType = "Unknown"
+)
+
+type InputPayload struct {
+	// The kubernetes resource that will be used as input
+	// +nullable
+	// +optional
+	KubeResource `json:",inline,omitempty"`
+}
+
+type InputResourceType string
+
+const (
+	InputResourceTypeKubeResource InputResourceType = "KubeGroupVersionResource"
+	InputResourceTypeUnknown      InputResourceType = "Unknown"
+)
+
+// ResourceInput defines a Kubernetes resource that will be fetched for CEL evaluation
+type KubeResource struct {
+	// Name is the variable name used to reference this resource in the CEL expression
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Type is the type of the resource
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=KubeGroupVersionResource
+	Type InputResourceType `json:"type"`
+
+	// APIGroup is the Kubernetes API group of the resource
+	// +kubebuilder:validation:Required
+	APIGroup string `json:"apiGroup"`
+
+	// Version is the Kubernetes API version of the resource
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Version string `json:"version"`
+
+	// Resource is the Kubernetes resource type
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Resource string `json:"resource"`
+
+	// Namespace is the Kubernetes namespace of the resource
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type CELPayload struct {
+
+	// The ID of the Rule
+	ID string `json:"id"`
+	// The title of the Rule
+	Title string `json:"title"`
+	// The description of the Rule
+	Description string `json:"description,omitempty"`
+	// The rationale of the Rule
+	Rationale string `json:"rationale,omitempty"`
+	// A discretionary warning about the of the Rule
+	Warning string `json:"warning,omitempty"`
+	// The severity level
+	Severity string `json:"severity,omitempty"`
+	// Instructions for auditing this specific rule
+	Instructions string `json:"instructions,omitempty"`
+	// What type of check will this rule execute:
+	// Platform or none (represented by an empty string)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Platform
+	CheckType string `json:"checkType,omitempty"`
+
+	// Expression is the CEL expression to evaluate
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Expression string `json:"expression"`
```

Comment:
```go
Evaluation (struct):
  // CELEvaluation
  Expression:
  Inputs:

  // BashEvaluation
  Commands: (list of command strings or reference to config map)

  // AnsibleEvaluation
  Tasks: (reference to config map or CRDs)

  // SomeNewThingWeDoNotKnowAboutYetEvaluation
```

Perhaps we can learn from https://github.com/openshift/cluster-logging-operator/blob/master/api/observability/v1/output_types.go